tokio/src/net/tcp/stream.rs

```diff
@@ -264,6 +264,33 @@ impl TcpStream {
         }
     }
 
+    /// Wait for any of the requested ready states.
+    pub async fn ready(&self, interest: Interest) -> io::Result<Ready> {
+        let event = self.io.registration().readiness(interest).await?;
+        Ok(event.ready)
+    }
+
+    /// Attempt a non-blocking read.
+    pub async fn try_read(&self, buf: &mut ReadBuf<'_>) -> io::Result<()> {
```

Comment:
It doesn't seem like this should be async?

Reply:
woops, you are right!